core/kernel/tee_misc.c

```diff
@@ -94,7 +94,7 @@ static bool is_valid_conf_and_notnull_size(
 		vaddr_t b, size_t bl, vaddr_t a, size_t al)
 {
 	/* invalid config return false */
-	if ((b + bl < b) || (a + al < a))
+	if ((b - 1 + bl < b) || (a - 1 + al < a))
```

Comment:
b = 4
bl = 4
What you want is this should be a valid configuration. But "b + b1 < b" will show you this is invalid.  we should use end address, but not end address plus 1. Here, bl = 4, the end address should be 7. Use "b - 1 + bl" can avoid such issue, and it can handle "b = 4, bl = 5".

On my platform, 
```
CFG_DDR_TEETZ_RESERVED_START ?= 0xfc100000
CFG_DDR_TEETZ_RESERVED_SIZE ?= 0x03f00000
```
base address is 0xfc100000, size is 0x3f00000, base + size will wrap down to 0, which is not what I want, because I want to use the top 64M for TEE, but b + bl will report this invalid.